tensorflow/core/ops/batchnorm_training_op.cc

```diff
@@ -0,0 +1,80 @@
+/* Copyright 2015 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/core/framework/op.h"
+
+namespace tensorflow {
+
+REGISTER_OP("BatchNormalizeTraining")
+    .Input("input: T")
+    .Input("scale: T")
+    .Input("bias: T")
+    .Input("running_mean: Ref(T)")
+    .Input("running_inv_var: Ref(T)")
+    .Output("out: T")
+    .Output("save_mean: T")
+    .Output("save_inv_var: T")
+    .Attr("T: {float}")
+    .Attr("epsilon: float")
+    .Attr("exponential_average_factor: float")
+    .Doc(R"doc(
+Perform batch normalization using batch statistics.
```

Comment:
TensorFlow default data_format is NHWC. So we need to support that at least. 

There are two ways: 

1. Call the Cudnn version of NHWC. 
2. Sometimes #1 is too slow, it is okay to call NHWCToNCHW first, do the op in NCHW, and then transform back to NHWC.